metaflow/client/core.py

```diff
@@ -1418,6 +1418,16 @@ def end_task(self):
 
         return end_step.task
 
+    @property
+    def exceptions(self):
+        """
+        Returns all the exceptions - if any - that occured during the execution.
+        """
+        exceptions = []
+        for step in self.steps():
+            exceptions.extend(task.exception for task in step.tasks() if task.exception)
```

Comment:
One minor nit here:
- maybe it would be nice to print the pathspec of the task as well (so maybe a dictionary keyed using the pathspec).

We should also investigate iterating over all tasks in a run. I don't think the API supports it directly right now though :(.